ui/src/app/job-details/resources/resources.component.ts

```diff
@@ -59,12 +59,17 @@ export class JobResourcesComponent implements OnInit {
       }
 
       if (this.job.extensions.logs) {
-        for (let file of Object.keys(this.job.extensions.logs)) {
-          this.readResourceFile(file);
-          let tabId = 'log-' + file;
-          this.tabIds.push(tabId);
-          this.tabTitles.set(tabId, file);
-        }
+        let files = Object.keys(this.job.extensions.logs);
+        Promise.all(files.map(file => this.readResourceFile(file)))
+          .then(entries => {
+            for (let [file, data] of entries.sort()) {
```

Comment:
How do string arrays get sorted? I think you'd probably want to just sort on key names.

Reply:
Per offline discussion, it will `toString` the tuples and sort that in ascending, ASCII character order.